WikiSentiment/WikiData/LanguageCollection.cs

```diff
@@ -0,0 +1,102 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace WikiSentiment.DataObjects
+{
+    //POJO for collection of articles in single language
+    public class LanguageCollection
+    {
+        public string countrycode { get; set; }
+
+        public int totalviews { get; set; }
+
+        public List<Article> articles { get; set; }
+
+        /// <summary>
+        /// Build daily collection of articles in single language
+        /// </summary>
+        /// <param name="_client">Http client for requests</param>
+        /// <param name="_date"></param>
+        /// <param name="_languageCode">Two letter code</param>
+        /// <param name="_exceptions">Dictionary with exceptions in {"en": ["Main_Page", "Search"]} format</param>
+        /// <param name="_keepMax">keep X amount of articles</param>
+        /// <returns></returns>
+        public static async Task<LanguageCollection> Create(
+            HttpClient _client, DateTime _date,
+            string _languageCode, Dictionary<string, string[]> _exceptions,
+            int _keepMax)
+        {
+            //get total views for a regional wikipedia
+            var totalviews = await WikiAPIRequests.GetTotalViews(_client, _languageCode, _date);
+
+            //get raw (title, views) data
+            var articleEntries = await WikiAPIRequests.GetArticleList(_client, _languageCode, _date);
+
+            var createdArticles = new Dictionary<string, Article>();
+
+            foreach (var iEntry in articleEntries)
+            {
+                //stop when created enough articles
+                if (createdArticles.Count >= _keepMax)
+                    break;
```

Comment:
```suggestion
            foreach (var iEntry in articleEntries.Take(_keepMax))
            {
```

Reply:
Well, you can ignore this, I didn't realize this was different collections.

You can theoretically do this with a `articleEntries.SelectMany(x => { loop contents }).Take(_keepMax)` but that's slightly hairy.